cmd/oceanbench/default.go

```diff
@@ -0,0 +1,135 @@
+/*
+NAME
+  This file contains default values used for variables across different
+	device types.
+
+AUTHORS
+  David Sutton <davidsutton@ausocean.org
+
+LICENSE
+  Copyright (C) 2018-2024 the Australian Ocean Lab (AusOcean)
+
+  This file is part of Ocean Bench. Ocean Bench is free software: you can
+  redistribute it and/or modify it under the terms of the GNU
+  General Public License as published by the Free Software
+  Foundation, either version 3 of the License, or (at your option)
+  any later version.
+
+  Ocean Bench is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  in gpl.txt.  If not, see
+  <http://www.gnu.org/licenses/>.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ausocean/cloud/model"
+	"github.com/ausocean/openfish/datastore"
+)
+
+// setupDefaultController takes the minimum settings required to define a Device and the settings store
+// and adds all the default variables, sensors, and actuators for a controller.
+//
+// Args:
+// - skey: Site Key of the new controller.
+// - name: Name of the device (Mutable).
+// - MAC: MAC address of the device optionally with colons (Immutable).
+// - SSID: SSID of the network for the controller to connect.
+// - password: password of the network.
+// - lat, long: latitude and longitude of the device.
+func setupDefaultController(ctx context.Context, skey int64, name, MAC, SSID, password string, lat, long float64, settingsStore datastore.Store) error {
+	// Check that the passed MAC is valid.
+	if !model.IsMacAddress(MAC) {
+		return model.ErrInvalidMACAddress
+	}
+
+	// Create the Controller with default settings.
+	controller := model.Device{
+		Skey:          skey,
+		Mac:           model.MacEncode(MAC),
+		Name:          name,
+		Inputs:        "A0,X50,X51,X60,X10",
+		Outputs:       "D14,D15,D16",
+		Wifi:          SSID + "," + password,
+		MonitorPeriod: 60,
+		ActPeriod:     60,
+		Type:          "Controller",
```

Comment:
I think this is worse, because this doesn't really link the changes in that other file to this one anyway. Ie if the index changes then this will change. But perhaps they can be handled better in set.go